src/net/protocol_agnostic/stream.rs

```diff
@@ -77,20 +92,25 @@ impl PaStream {
         <C as Sink>::SinkError: fmt::Debug,
         C: 'static,
     {
+        let disable_tcp = match config.read().dev {
```

Comment:
Duplicate code with https://github.com/maidsafe/crust/pull/903/commits/78c87256506593092e8075d6ac8e39d72d6f17df#diff-2769f48124e6f45bc5d555b5c1d61f29R55

Couldn't this be some function of config, e.g. `config.tcp_enabled() -> bool`